tsdb/importer/import_test.go

```diff
@@ -0,0 +1,753 @@
+// even if the other labels b/w the metrics are different.
+func readSeries(block tsdb.BlockReader, lbls labels2.Labels) ([]tsdb.MetricSample, error) {
+	series := make([]tsdb.MetricSample, 0)
+	ir, err := block.Index()
+	if err != nil {
+		return series, err
+	}
+	defer ir.Close()
+	tsr, err := block.Tombstones()
+	if err != nil {
+		return series, err
+	}
+	defer tsr.Close()
+	chunkr, err := block.Chunks()
+	if err != nil {
+		return series, err
+	}
+	defer chunkr.Close()
+	for _, lbl := range lbls {
+		css, err := tsdb.LookupChunkSeries(ir, tsr, labels2.MustNewMatcher(labels2.MatchEqual, lbl.Name, lbl.Value))
+		if err != nil {
+			return series, err
+		}
+		for css.Next() {
+			actLabels, chkMetas, _ := css.At()
+			var chunkIter chunkenc.Iterator
+			for _, meta := range chkMetas {
+				chk, err := chunkr.Chunk(meta.Ref)
+				if err != nil {
+					return series, err
+				}
+				chunkIter = chk.Iterator(chunkIter)
+				for chunkIter.Next() {
+					t, v := chunkIter.At()
+					sample := tsdb.MetricSample{
+						Timestamp: t,
+						Value:     v,
+						Labels:    actLabels,
+					}
+					series = append(series, sample)
+				}
+			}
+		}
+	}
+	return series, nil
+}
+
+// genSeries generates series from mint to maxt, with a step value.
+func genSeries(labels []string, mint, maxt int64, step int) []tsdb.MetricSample {
+	series := make([]tsdb.MetricSample, 0)
+	for idx := mint; idx < maxt; idx += int64(step) {
+		// Round to 3 places.
+		val := math.Floor(rand.Float64()*1000) / 1000
+		sample := tsdb.MetricSample{Timestamp: idx, Value: val, Labels: labels2.FromStrings(labels...)}
+		series = append(series, sample)
+	}
+	return series
+}
+
+// labelsToStr converts the given labels to a string representation that is compliant with
+// the OpenMetrics parser.
+func labelsToStr(labels labels2.Labels) string {
+	str := "{"
+	for idx, l := range labels {
+		str += fmt.Sprintf("%s=%s", l.Name, strconv.Quote(l.Value))
+		if idx < len(labels)-1 {
+			str += ","
+		}
+	}
+	str += "}"
+	return str
+}
+
+// genOpenMetricsText formats the given series data into OpenMetrics compliant format.
+func genOpenMetricsText(metricName, metricType string, series []tsdb.MetricSample) string {
+	str := fmt.Sprintf("# HELP %s This is a metric\n# TYPE %s %s", metricName, metricName, metricType)
+	for _, s := range series {
+		str += fmt.Sprintf("\n%s%s %f %d", metricName, labelsToStr(s.Labels), s.Value, s.Timestamp)
+	}
+	str += fmt.Sprintf("\n# EOF")
+	return str
+}
+
+func TestImport(t *testing.T) {
+	tests := []struct {
+		ToParse      string
+		IsOk         bool
+		MetricLabels []string
+		Expected     struct {
+			MinTime   int64
+			MaxTime   int64
+			NumBlocks int
+			Symbols   []string
+			Samples   []tsdb.MetricSample
+		}
+	}{
+		{
+			ToParse: `# EOF`,
+			IsOk:    true,
+		},
+		{
+			ToParse: `# HELP http_requests_total The total number of HTTP requests.
+# TYPE http_requests_total counter
```

Comment:
Nice, but I am missing test that actually test import of OpenMetircs/Prom exposition with multiple samples :thinking: 
Is this supported?

Reply:
Yes, the test named `TestImportIntoExistingDB` is for exactly that case, where we import multiple samples into the DB.

Reply:
Why not having this in the same TableTest? Why complicating? (:

Reply:
Also I checked that test

```
// genOpenMetricsText formats the given series data into OpenMetrics compliant format.
func genOpenMetricsText(metricName, metricType string, series []tsdb.MetricSample) string {
	str := fmt.Sprintf("# HELP %s This is a metric\n# TYPE %s %s", metricName, metricName, metricType)
	for _, s := range series {
		str += fmt.Sprintf("\n%s%s %f %d", metricName, labelsToStr(s.Labels), s.Value, s.Timestamp)
	}
	str += fmt.Sprintf("\n# EOF")
	return str
}
```

This is not allowing me to inject samples just single samples, no? 

So how do user import series with let's say 100 samples? (: